Initialization + simple Python wrapper

dkobak opened the pull request:
I have implemented initialization. It works as follows: if there are N*no_dims doubles in the input file after the seed is read, then these are put into `Y`. Otherwise, no initialization is used. So all existing code  should keep working (full backwards compatibility).

As I am working in Python, I have also written a simple Python wrapper, which is very similar to your Matlab/R wrappers. I've seen that there is a third-party Python wrapper that I think uses Cython, but I guess it can be useful to have a very simple Python wrapper directly in here.

The Python wrapper supports initialization. I did *NOT* implement initialization support in the Matlab/R wrappers. Maybe it's better if you do it.

linqiaozhi commented:
Looks pretty great, thanks so much @dkobak!  I think it would be best to first implement this in the Matlab, R, and Cython wrappers, before I merge--this way everything is consistent. It might take me a few days until I can get around to doing that, but this is definitely a great contribution. 

Thanks again.

dkobak commented:
OK, let me know if you prefer me to update the Matlab and the R wrappers. You'd only need to double-check that they work as intended.

The Cython wrapper is not in this repository so I don't know what to do about that one. I am currently using my simple Python wrapper that is part of this pull request.

linqiaozhi commented:
If you'd be willing to update the Matlab and R wrappers that would be a big help--I can merge it when you're done. Then I can update the Cython wrapper myself (probably early next week). 

Thanks again for your help with this!!

dkobak commented:
* I added initialization support to Matlab and R wrappers.
* I also added K and sigma support to Matlab and R wrappers (they did not allow setting them for some reason).
* Also, I got rid of the `Ignoring fread() output` warnings -- they were too annoying :)
* Also, now fixed sigma case is correctly handled by both Annoy and VP algorithms.

Warning: I did not test Matlab and R wrappers. The additions were rather trivial so I hope I did not break anything, but it would be better to double-check.

I will pull request all of that now.
